python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_auto_materialize_asset_evaluation.py

```diff
@@ -100,6 +100,11 @@ def test_backcompat():
         num_discarded=0,
         dynamic_partitions_store=None,
     )
+
+    # This field is populated via the auto materialize policies on the asset graph,
+    # replace to be None
+    expected_asset_evaluation.rule_snapshots._replace(rule_snapshots=None)
```

Comment:
Does `_replace` mutate the object? I thought it just returns a new object and doesn't modify the underlying object. So this line wouldn't have any effect?